#! /bin/sh -e
## 52_check_perms_lstat.dpatch by Tollef Fog Heen <tfheen@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: ?

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch ${2:+-d $2}}"

if [ $# -lt 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $patch_opts -p1 < $0;;
    -unpatch) patch $patch_opts -p1 -R < $0;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1;;
esac

exit 0
@DPATCH@
--- mailman-2.1.4.orig/bin/check_perms
+++ mailman-2.1.4/bin/check_perms
@@ -81,7 +81,7 @@
     return os.stat(path)[ST_MODE]
 
 def statgidmode(path):
-    stat = os.stat(path)
+    stat = os.lstat(path)
     return stat[ST_MODE], stat[ST_GID]
 
 seen = {}
